packages/frontend/src/hooks/useSqlQueryVisualization.ts

```diff
@@ -0,0 +1,124 @@
+import {
+    ApiQueryResults,
+    ChartType,
+    CompiledDimension,
+    DimensionType,
+    Explore,
+    fieldId,
+    FieldId,
+    FieldType,
+    friendlyName,
+    SupportedDbtAdapter,
+} from '@lightdash/common';
+import moment from 'moment';
+import { useMemo, useState } from 'react';
+import { useSqlQueryMutation } from './useSqlQuery';
+
+const SQL_RESULTS_TABLE_NAME = 'sql_runner';
+
+type Args = {
+    sqlQueryMutation: ReturnType<typeof useSqlQueryMutation>;
+};
+
+const useSqlQueryVisualization = ({ sqlQueryMutation: { data } }: Args) => {
```

Comment:
This hook, based on the sql query results:
- generates a mock explore
- generates a mock metric query
- detects the dimension types based on the values from the first results row. This isn't perfect since the values might not represent the entire data set but is a good first step. Open to other suggestions.